cap-ci/pipeline.template

```diff
@@ -87,99 +102,135 @@ resources:
 
 {{ $pipelineName := .Env.PIPELINE }}
 
+{{- $create_k8s := `
+            # create k8s cluster
+            export random_variable=$(hexdump -n 8 -e '2/4 "%08x"' /dev/urandom)
+            export AWS_ACCESS_KEY_ID='((aws-ci-chuller-access-key-id))'
+            export AWS_SECRET_ACCESS_KEY='((aws-ci-chuller-secret-access-key))'
+            if [[ ${BACKEND} == "gke" ]]; then
+              printf "%s" '((gke-key-json))' > $PWD/gke-key.json
+              export GKE_CRED_JSON=$PWD/gke-key.json
+              export GKE_CLUSTER_NAME=ci-${BACKEND}-${random_variable}
+              export CLUSTER_NAME=${GKE_CLUSTER_NAME}
+              export TF_KEY=${GKE_CLUSTER_NAME}
+              (cd catapult; make BACKEND=$BACKEND GKE_CREDS_JSON=$GKE_CRED_JSON GKE_CLUSTER_NAME=${GKE_CLUSTER_NAME} CLUSTER_NAME=${CLUSTER_NAME} TF_KEY=${TF_KEY} k8s)
+              echo ${GKE_CLUSTER_NAME} > kubeconfig-pool/name
+              cp catapult/build${GKE_CLUSTER_NAME}/kubeclusterreference kubeconfig-pool/metadata
+              echo tf-${GKE_CLUSTER_NAME}.zip > tfstate-pool/name
+              cp catapult/build${GKE_CLUSTER_NAME}/tf-setup.zip tfstate-pool/metadata
+            else
+              echo "CI cannot create ${BACKEND} cluster"
+              exit 1
+            fi
+` }}
+
+{{- $delete_k8s := `
+                # delete k8s cluster
+                cp ../tfstate-pool/metadata tfstate.zip
+                export TFSTATE=$PWD/tfstate.zip
+                export AWS_ACCESS_KEY_ID='((aws-ci-chuller-access-key-id))'
+                export AWS_SECRET_ACCESS_KEY='((aws-ci-chuller-secret-access-key))'
+                if [[ ${BACKEND} == "gke" ]]; then
+                    printf "%s" '((gke-key-json))' > $PWD/gke-key.json
+                    export GKE_CRED_JSON=$PWD/gke-key.json
+                    export SCF_OPERATOR=true
+                    make clean
+                fi
+` }}
+
 {{- $obtain_kubeconfig := `
-            # obtain kubeconfig from pool
-            pool_file=${pool_file:-pool.kube-hosts/metadata}
-            export CLUSTER_NAME=$(cat pool.kube-hosts/name)
-            cp ${pool_file} catapult/kubeconfig_$CLUSTER_NAME
-            pushd catapult
+                # obtain kubeconfig from pool
```

Comment:
Any reason for introducing tabs? same comment for rest of the tabs.

Reply:
yes. yaml spacing issues.